tests/main/interfaces-hardware-control/task.yaml

```diff
@@ -0,0 +1,41 @@
+summary: Ensure that the hardware-control interface works.
+
+summary: |
+    The hardware-control interface allows a snap to control hardware and platform devices.
+
+    A snap which defines the hardware-control plug must be shown in the interfaces list.
+    The plug must not be connected on install and, as usual, must be able to be
+    reconnected.
+
+    A snap declaring a plug on this interface must be able to write files in /sys/{block,bus,class,devices}
+
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB/snaps.sh"
+
+    echo "Given a snap declaring a plug on the hardware-control interface is installed"
+    install_local hardware-control-consumer
+
+execute: |
+    echo "The interface is not connected by default"
+    snap interfaces -i hardware-control | MATCH '^- +hardware-control-consumer:hardware-control'
+
+    echo "When the plug is connected"
+    snap connect hardware-control-consumer:hardware-control
+
+    echo "Then the snap is able to read hardware information"
+    hardware-control-consumer.consumer
+
+    if [ "$(snap debug confinement)" = partial ] ; then
+        exit 0
+    fi
+
+    echo "When the plug is disconnected"
+    snap disconnect hardware-control-consumer:hardware-control
+
+    echo "Then the snap is not able to write the hardware information"
+    if hardware-control-consumer.consumer 2>hw.error; then
+        echo "Expected permission error accessing locale configuration with disconnected plug"
+        exit 1
+    fi
+    grep -q "Permission denied" hw.error
```

Comment:
Could you please use `MATCH` for this. It behaves like grep but provides a nicer error on failure.

Reply:
No problem, have updated the test.